### PR TITLE
Add Universal.murmurHash

### LINKED ISSUE #3756: `Universal.murmurHash : a -> Nat` builtin
Handy for implementing hash tables.

`Universal.murmurHashBytes : Bytes -> Nat`

### PR DESCRIPTION
**Pull request by SystemFw**
## Overview

Add a universal non crypto hash function, useful for hash map. Fixes #3756 

## Implementation notes

Uses Haskell's `murmur-hash`, which already integrates with ByteString

## Interesting/controversial decisions

None

## Test coverage

I've done some manual testing, and the code is a very thin wrapper over the haskell lib.

## Loose ends

Not sure whether it's worth adding a `murmurBytes: Bytes -> Nat` as well or not, it can currently be written in pure Unison

### CODE REVIEW

**Review by pchiusano — APPROVED** [review]
Nice! I'd just add a transcript test to builtins.md to exercise it, otherwise looks good.

**Comment by SystemFw** [comment]
`murmurHash` seems to be only be deterministic within a ucm session without caching. The results are different in different ucm sessions. The issue seems to be with `serializeValueLazy` (assuming the haskell library is correct)

File: unison-src/transcripts/builtins.md [hunk 1]
```diff
@@ -356,6 +356,19 @@ openFile]
 .> add
 ```
 
+## Universal hash functions
+
+Just exercises the function
+
+```unison
+> Universal.murmurHash 1
+test> checks [Universal.murmurHash [1,2,3] == Universal.murmurHash [1,2,3]]
```
Comment (pchiusano) [inline unison-src/transcripts/builtins.md]: Make sure you give the test a name here - 

```suggestion
test> Universal.murmurHash.tests = checks [Universal.murmurHash [1,2,3] == Universal.murmurHash [1,2,3]]
```

**Comment by pchiusano** [comment]
For CI to pass this just needs a merge from trunk and a transcript refresh. I'm applying ready-to-merge and it'll merge to trunk once CI passes.

**Pull request merged**